background/services/keyring/index.ts

```diff
@@ -352,7 +352,30 @@ export default class KeyringService extends BaseService<Events> {
       }
 
       this.#keyrings.push(newKeyring)
-      const [address] = newKeyring.addAddressesSync(1)
+      //const [address] = newKeyring.addAddressesSync(1)
+
+      // FIXME temp solution for SDK v5
+      // create a new address until we find an address for the target shard
+      let address
+      let found = false
+      const DEFAULT_SHARD = "cyprus-1"
+      while (!found) {
+        address = newKeyring.addAddressesSync(1)[0]
+        const shardFromAddress = getShardFromAddress(address)
+        if (shardFromAddress !== undefined) {
+          if (shardFromAddress === DEFAULT_SHARD) {
+            found = true
+            break
+          }
+        }
+        this.#hiddenAccounts[address] = true // may want to reconsider this
```

Comment:
Not sure if using hidden account is the best approach here but I am going to go ahead and approve this PR as the behavior is correct after testing.